### PR TITLE
[Feat] Add elastic transform 2d

### PR DESCRIPTION
**Pull request by edgarriba**
### Description

Implements the elastic transform functional based on #471.

### Status
**Ready**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [ ] Did you discuss the functionality or any breaking changes before ?
- [ ] **Pass all tests**: did you test in local ? `make test`
- [ ] Unittests: did you add tests for your new functionality ?
- [ ] Documentations: did you build documentation ? `make build-docs`
- [ ] Implementation: is your code well commented and follow conventions ? `make lint`
- [ ] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>
  
  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?
 
</details>

### CODE REVIEW

**Comment by edgarriba** [comment]
@IssamLaradji please check. /cc @anguelos in case you want to review, I made functional as you suggested.

**Comment by IssamLaradji** [comment]
@edgarriba looks very good to me. Does the current Kornia `gaussian` function support differentiation? I remember I had to add this function to allow the gradients to follow,

```
def gaussian(window_size, sigma):
    r"""
    Modified from Kornia to allow gradients to flow
    """
    x = torch.arange(window_size).float().to(
        device=sigma.device) - window_size // 2
    if window_size % 2 == 0:
        x = x + 0.5
    gauss = torch.exp((-x.pow(2.0) / (2 * sigma ** 2)))
    return gauss / gauss.sum()
```

**Comment by edgarriba** [comment]
@IssamLaradji you were right. It's fixed now. Might be unlikely but shouldn't we add an epsilon value in the exponential in case the optimizer throws an absolute zero ?

**Comment by IssamLaradji** [comment]
ah good point about the sigma being zero @edgarriba. I am not sure whether we need the epsilon, because I feel that the optimizer will always have a value that is not exactly zero! But I will keep an eye on this if the issue ever happens.
